### DISCUSSION TITLE
Can't send and recive emails

### DISCUSSION THREAD

**Discussion by Colors45** [modoboa/modoboa#2734] [Help]
I configured a domain and added the requiered DNS Records.
![image](https://user-images.githubusercontent.com/120259605/210226141-143fdaa6-6792-4a27-97b1-66153da338ab.png)

When I no login into the webmail I can't recive and also send emails. 
In the logs I found this error:
```
postfix/postscreen[3596784]: warning: cidr:/etc/postfix/postscreen_spf_whitelist.cidr is unavailable. open /etc/postfix/postscreen_spf_whitelist.cidr: No such file or directory
postfix/postscreen[3596784]: warning: postscreen_access_list: cidr:/etc/postfix/postscreen_spf_whitelist.cidr: table lookup error -- ignoring the remainder of this access list
...
postfix/smtpd[3596796]: warning: connect to 127.0.0.1:9999: Connection refused
postfix/smtpd[3596796]: warning: problem talking to server 127.0.0.1:9999: Connection refused
postfix/smtpd[3596796]: warning: connect to 127.0.0.1:9999: Connection refused
postfix/smtpd[3596796]: warning: problem talking to server 127.0.0.1:9999: Connection refused
postfix/smtpd[3596796]: NOQUEUE: reject: RCPT from <Sender Domain>[<IP 1>]: 451 4.3.5 <My Maildomain>: Recipient address rejected: Server configuration problem;
```

Installer: ✅ 
Version: `2.0.3`

Hope someone can help me to fix this issue

**Comment by Spitfireap** [comment]
Hi,
You seem to have 2 issues:
1. Postscreen issue : this is related to a known bug in the installer, a fix is underway. To fix it manually : `cp /usr/local/bin/postwhite/postwhite.conf /etc && /usr/bin/postwhite/postwhite`.
2. Second issue is the policy daemon. Please check `netstat -tlnup | grep 9999`. If it is empty : `cat /var/log/supervisor/supervisord.log` and check that policy daemon is not restarting every second.

**Reply by Colors45** [reply]
`cp /usr/local/bin/postwhite/postwhite.conf /etc && /usr/bin/postwhite/postwhite` give me this error: `/usr/bin/postwhite/postwhite: No such file or directory`


`netstat -tlnup | grep 9999` gives me:
```
tcp        0      0 0.0.0.0:19999           0.0.0.0:*               LISTEN      2392025/netdata
tcp6       0      0 :::19999                :::*                    LISTEN      2392025/netdata
```

**Reply by Spitfireap** [reply]
Yes, it's a typo : `/usr/local/bin/postwhite/postwhite`.

So policy daemon is not started, please check the log of supervisord.

**Reply by Colors45** [reply]
In the log the following lines are spammed:
```
2023-01-02 13:45:06,312 INFO spawned: 'policyd' with pid 3872968
2023-01-02 13:45:07,315 INFO success: policyd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-01-02 13:45:08,192 INFO exited: policyd (exit status 1; not expected)
```

The only thing that changes is the pid, the rest stays the same

**Reply by Spitfireap** [reply]
what is the output of `ifconfig  | grep ::1`

**Reply by Colors45** [reply]
```
inet6 fdba:17c8:6c94::1011  prefixlen 64  scopeid 0x0<global>
inet6 fe80::1  prefixlen 64  scopeid 0x20<link>
```

**Reply by Spitfireap** [reply]
and `cat /etc/hosts | grep localhost` ?

**Reply by Colors45** [reply]
```
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
```

**Reply by Spitfireap** [reply]
> ```
> ::1     localhost ip6-localhost ip6-loopback
> ```

Can you comment this line and restart? Then check supervisord log.

**Reply by Colors45** [reply]
I now can recive emails but not send them.
The request in the browser returns 500:

Logs:
```
Jan  2 15:49:33 Server postfix/submission/smtpd[53892]: connect from <Domain of my Server>[<IP of my Server>]
Jan  2 15:49:33 Server postfix/submission/smtpd[53892]: lost connection after UNKNOWN from <Domain of my Server>[<IP of my Server>]
Jan  2 15:49:33 Server postfix/submission/smtpd[53892]: disconnect from <Domain of my Server>[<IP of my Server>] unknown=0/1 commands=0/1
```

**Reply by Spitfireap** [reply]
Are you using the webmail ? If so, have you changed webmail preferences from a SuperAdmin account ? Could you try using an SMTP client such as thunderbird ?

**Reply by Colors45** [reply]
When I use Thunderbird the email get's stuck at sending.
In the logs I see `SSL_accept error`, `TLS library problem` and `lost connection after STARTTLS`

**Reply by PatTheMav** [reply]
Do you get any errors from running `openssh s_client -connect <YOUR_MAIL_SERVER_DOMAIN>:25 -starttls smtp` (do **not** paste its output into a reply)?

To use `STARTTLS` your server needs to be properly configured with TLS certificates generated for your specific domain and configured for `postfix` (I haven't used the installer in years, so I don't recall if it fully takes care of that).

**Reply by Colors45** [reply]
Openssh isn't installed on my system: `Command 'openssh' not found`
I have valid(and active) TLS certificate(a free Let's Encrypt certificate created with certbot) for the domain that nginx actively uses.

**Reply by PatTheMav** [reply]
You can use `openssh` from your client machine as well, it doesn't need to be done on the server.

**Reply by Colors45** [reply]
On which operating system should the command work? I check windows and ubuntu. The command didn't work on both systems.

**Reply by PatTheMav** [reply]
That's because I'm stupid - the correct command is `openssl`, not `openssh` - auto correct on the Mac is weird sometimes..

**Reply by Colors45** [reply]
I don't get an error, but it get's stuck at `read R BLOCK`

**Reply by PatTheMav** [reply]
That is correct, you opened a session with the server and it's awaiting commands/data - you can type `QUIT` to end the SMTP session.

If the `SSL-Session` says `Verify return code: 0 (ok)`, then TLS negotiation was successful.

**Reply by Colors45** [reply]
When I type `QUIT` I get `DONE`

**Reply by Colors45** [reply]
I still didn't got it fixed

**Reply by Colors45** [reply]
Is there anyone here who can help me please? It's annoying that I can't send emails and it's been like this for almost a week now.
